lib/_http_outgoing.js

```diff
@@ -728,8 +728,23 @@ OutgoingMessage.prototype.setHeaders = function setHeaders(headers) {
     throw new ERR_INVALID_ARG_TYPE('headers', ['Headers', 'Map'], headers);
   }
 
-  for (const key of headers.keys()) {
-    this.setHeader(key, headers.get(key));
+  // Headers object joins multiple cookies with a comma when using
+  // the getter to retrieve the value,
+  // unless iterating over the headers directly.
+  // We also cannot safely split by comma.
+  // To avoid setHeader owerwriting the previous value we push
+  // set-cookie values in array and set them all at once.
+  const cookies = [];
+
+  for (const { 0: key, 1: value } of headers.entries()) {
```

Comment:
Alternatively you could ignore headers where the key is `set-cookie` and then use headers.getSetCookie

Reply:
```suggestion
  for (const { 0: key, 1: value } of headers) {
```

is more apropriate in my vision

Reply:
@KhafraDev the value could be a map so might not be available `getSetCookie`